app/models/course/leaderboard_settings.rb

```diff
@@ -35,13 +38,45 @@ def display_user_count
     @settings.display_user_count || 30
   end
 
-  # Returns the number of users to be displayed on the leaderboard
+  # Set the number of users to be displayed on the leaderboard
   #
   # @param [Integer] count The number of users to be displayed
   def display_user_count=(count)
     @settings.display_user_count = count
   end
 
+  # Returns whether group leaderboard is enabled (disabled by default).
+  #
+  # @return [Boolean] Setting on whether group leaderboard is enabled.
+  def enable_group_leaderboard
+    @group_leaderboard_settings.settings(:enabled) == true
+  end
+
+  # Enable or disable the option to display group leaderboard
+  #
+  # @param [Boolean|Fixnum|String] option Setting on whether group leaderboard is enabled.
+  #   By default, simple_form provides '0' and '1' for boolean fields.
+  #   This method will handle this conversion to Boolean.
+  def enable_group_leaderboard=(option)
+    option = TRUE_VALUES.include?(option)
```

Comment:
Probably then this should be a library extension. I don't think this would be the only place where we'd use this functionality.